src/main/java/io/confluent/connect/jdbc/JdbcSourceTask.java

```diff
@@ -145,6 +149,35 @@ public void start(Map<String, String> properties) {
         default:
           throw new ConnectException("Unexpected query mode: " + queryMode);
       }
+
+      String keyColumn = null;
+      ResultSet primaryKeys = null;
+
+      try {
+        primaryKeys = db.getMetaData().getPrimaryKeys(null, null, tableOrQuery);
+
+        if (primaryKeys.next()) {
+          String candidateKeyColumn = primaryKeys.getString(4);
+
+          if (!primaryKeys.next()) {
+            keyColumn = candidateKeyColumn;
+          } else {
+            log.warn("Composite primary key found in table {}, not supported", tableOrQuery);
```

Comment:
Yeah I'd argue that it's just a warning, since in general I don't think the JDBC driver should reject tables that have composite primary keys, it can simply fall back on the behavior it has now, where no key is specified for the record -- I will make sure there are appropriate errors if you specify the primary key as the incrementing column but the table does not have a primary key or has a composite primary key.